src/App.ts

```diff
@@ -7,83 +7,193 @@ import * as bodyParser from "body-parser";
+    redisClient,
+    config.PAYMENT_ACTIVATION_STATUS_TIMEOUT_SECONDS
+  );
+  await attachFespCdServer(
+    app,
+    CONFIG.CONTROLLER.ROUTES.SOAP.PAYMENT_ACTIVATIONS_STATUS_UPDATE,
+    fespCdServiceHandler
+  );
+
   const server = http.createServer(app);
   server.listen(config.CONTROLLER.PORT);
+
   logger.info(
     `Server started at ${config.CONTROLLER.HOST}:${config.CONTROLLER.PORT}`
   );
   return server;
 }
 
 export function stopServer(server: http.Server): void {
-  logger.info("Stopping Proxy PagoPa Server...");
+  logger.info("Stopping Proxy PagoPA Server...");
   server.close();
 }
 
-// Set Restful WS routes
+/**
+ * Set RESTful WS endpoints
+ * @param {core.Express} app - Express server to set
+ * @param {Configuration} config - PagoPa proxy configuration
+ * @param {redis.RedisClient} redisClient - The redis client used to store information sent by PagoPA
+ * @param {PagamentiTelematiciPspNodoAsyncClient} pagoPAClient - PagoPa SOAP client to call verificaRPT and attivaRPT services
+ */
 function setRestfulRoutes(
   app: core.Express,
   config: Configuration,
-  verificaRPTPagoPaClient: pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient,
-  attivaRPTPagoPaClient: pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient
+  redisClient: redis.RedisClient,
+  pagoPAClient: pagoPASoapClient.PagamentiTelematiciPspNodoAsyncClient
 ): void {
+  const jsonParser = bodyParser.json();
+
+  const getPaymentInfoHandler = toExpressHandler(
+    PaymentController.getPaymentInfo(config.PAGOPA, pagoPAClient)
+  );
   app.get(
-    config.CONTROLLER.ROUTES.RESTFUL.PAYMENTS_CHECK,
-    (req: express.Request, res: express.Response) => {
-      logger.info("Serving Payment Check Request (GET)...");
-      toExpressHandler(
-        PaymentController.checkPaymentToPagoPa(
-          config.PAGOPA,
-          verificaRPTPagoPaClient
-        )
-      )(req, res);
-    }
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_REQUESTS_GET,
+    getPaymentInfoHandler
+  );
+
+  const activatePaymentHandler = toExpressHandler(
+    PaymentController.activatePayment(config.PAGOPA, pagoPAClient)
   );
   app.post(
-    config.CONTROLLER.ROUTES.RESTFUL.PAYMENTS_ACTIVATION,
-    (req: express.Request, res: express.Response) => {
-      logger.info("Serving Payment Activation Request (POST)...");
-      toExpressHandler(
-        PaymentController.activatePaymentToPagoPa(
-          config.PAGOPA,
-          attivaRPTPagoPaClient
-        )
-      )(req, res);
-    }
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_ACTIVATIONS_POST,
+    jsonParser,
+    activatePaymentHandler
+  );
+
+  const getActivationStatusHandler = toExpressHandler(
+    PaymentController.getActivationStatus(redisClient)
   );
+  app.get(
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_ACTIVATIONS_GET,
+    getActivationStatusHandler
+  );
+
+  // Endpoint for OpenAPI handler
+  app.get("/specs/api/v1/swagger.json", GetOpenapi(publicApiV1Specs));
```

Comment:
why `/specs` ? `/api/v1/swagger.json` is fine - you want [the same base path defined in the specs](https://github.com/teamdigitale/italia-pagopa-proxy/pull/46/files#diff-53ce2f86f6ea92447b5e66be77055ca6R6)